component/GroceriesList/index.js

```diff
@@ -8,30 +8,77 @@ import {
 } from "./GroceriesList.styled.js";
 import { calculateDaysRemaining } from "@/resources/calculateDaysRemaining.js";
 
-export default function GroceriesList() {
+export default function GroceriesList({ filterValue }) {
+  const filteredGroceriesList = groceries.filter((item) => {
+    const currentDate = new Date();
+    const expirationDate = new Date(item.expirationDate);
+    const remainingDays = Math.ceil(
+      (expirationDate - currentDate) / (1000 * 60 * 60 * 24)
+    );
+
+    if (!filterValue || filterValue === -1) {
+      return (
+        <StyledList>
+          {groceries.map(
+            ({ id, emoji, name, purchasedDate, expirationDate }) => (
+              <StyledListContainer key={id}>
+                <StyledHeading className="itemEmoji">{emoji}</StyledHeading>
+                <StyledDetailSection>
+                  <p className="itemName">Name: {name}</p>
+                  <p className="itemPurchasedDate">
+                    Purchased Date: {purchasedDate}
+                  </p>
+                  <p className="itemExpirationDate">
+                    Expiration Date: {expirationDate}
+                  </p>
+                </StyledDetailSection>
+                <StyledRemainingDays className="remainingDays">
+                  Expire in: {calculateDaysRemaining(expirationDate)} days
+                </StyledRemainingDays>
+              </StyledListContainer>
+            )
+          )}
+        </StyledList>
+      );
+    }
```

Comment:
You are accidentally using JSX within the filter function. If you look closely, you'll see that the filter function extends to line 52. That's where it ends. Until then, the filter function only needs to know whether or not to include an item.

Reply:
Hey, thank you so much for review my code and very helpful feedback. I pushed it again and did what you suggested :)